python/tests/test_writer.py

```diff
@@ -501,3 +501,40 @@ def test_writer_with_options(tmp_path: pathlib.Path):
     )
 
     assert table == data
+
+
+def test_try_get_table_and_table_uri(tmp_path: pathlib.Path):
+    data = pa.table({"vals": pa.array(["1", "2", "3"])})
+    table_or_uri = str(tmp_path / "delta_table")
+    write_deltalake(table_or_uri, data)
+    delta_table = DeltaTable(table_or_uri)
```

Comment:
I see. We need `DeltaTable.__init__` to support `Path` too. Would you be willing to add that as well?

Reply:
Added it there! And also in `try_get_deltatable`